lib/IMAP/MailboxSync.php

```diff
@@ -120,19 +121,21 @@ private function updateMailboxFromFolder(Folder $folder, Mailbox $mailbox): void
 		$mailbox->setMessages(0); // TODO
 		$mailbox->setUnseen(0); // TODO
 		$mailbox->setSelectable($folder->isSelectable());
+		$mailbox->setSpecialUse(json_encode($folder->getSpecialUse()));
 		$this->mailboxMapper->update($mailbox);
 	}
 
 	private function createMailboxFromFolder(Account $account, Folder $folder): void {
 		$mailbox = new Mailbox();
-		$mailbox->setId($folder->getMailbox());
+		$mailbox->setName($folder->getMailbox());
 		$mailbox->setAccountId($account->getId());
 		$mailbox->setSyncToken($folder->getSyncToken());
 		$mailbox->setAttributes(json_encode($folder->getAttributes()));
 		$mailbox->setDelimiter($folder->getDelimiter());
 		$mailbox->setMessages(0); // TODO
 		$mailbox->setUnseen(0); // TODO
 		$mailbox->setSelectable($folder->isSelectable());
+		$mailbox->setSpecialUse(json_encode($folder->getSpecialUse()));
```

Comment:
Do we know beforehand what the special uses can be? As in how long this final string can get?

Reply:
A combination of https://github.com/nextcloud/mail/blob/f40aba36d6356acb23cd0fb6aa9e6584f69fdca3/lib/IMAP/FolderMapper.php#L180-L186, which are all ~4-6 chars long. And usually a folder has just one of these special uses.

The max length shouldn't be a problem.